src/test/UserServiceTest.java

```diff
@@ -0,0 +1,133 @@
+
+    @Test
+    public void testAddUser_addsUser()
+    {
+        fixture.givenUserIsInitialized("Ahmed",10,0);
+        fixture.whenUserIsAddedSuccessfully();
+        fixture.thenAssertionResultsInEquality(1);
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void testAddUser_ageIsNegative_doesNotAddUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitialized("Ahmed",-10,0);
+            fixture.whenUserIsNotAddedSuccessfully();
+        }catch (IllegalArgumentException e)
+        {
+            fixture.thenAssertionResultsInEquality(0);
+            fixture.thenThrowIllegalArgumentException();
+        }
+    }
+
+    @Test
+    public void testRemoveUser_removesUser()
+    {
+        fixture.givenUserIsInitializedAndAdded("Ahmed",10,1);
+        fixture.whenUserIsDeleted();
+        fixture.thenAssertionResultsInEquality(0);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testRemoveUser_userDoesNotExist_doesNotRemoveUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitializedAndAdded("Ali",10,0); // User count=1
+            fixture.givenUserIsReInitializedAfterAddition("Ahmed",11,1); // assigning same object to a new reference
+            fixture.whenUserIsDeleted();  // will look for Ahmed in usersArray, won't find it so won't delete, count remains 1
+        }catch (IndexOutOfBoundsException e){
+            fixture.thenAssertionResultsInEquality(1);
+            fixture.thenThrowIndexOutOfBoundException();
+        }
+    }
+
+    @Test
+    public void testEditUser_editsUserSuccessfully()
+    {
+        fixture.givenUserIsInitializedAndAdded("Ahmed",10,0);
+        fixture.whenUserIsEdited("Ali",20);
+        fixture.thenAssertEqualityCheckOnEditUser();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEditUser_enteredInvalidAttributes_doesNotEditUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitializedAndAdded("Ahmed",10,0);
+            fixture.whenUserIsEdited("",0);
+        } catch(IllegalArgumentException e)
+        {
+            fixture.thenAssertEqualityCheckOnEditUser();
+            fixture.thenThrowIllegalArgumentException();
+        }
+    }
+}
+class Fixture{
+    private UserService userService = new UserService();
+    private int expectedCount;
+    private User userMock=Mockito.mock(User.class);
+    public void givenUserIsInitialized(String name, int age, int id) {
+        this.userMock.setName(name);
+        this.userMock.setAge(age);
+        this.userMock.setId(id);
+    }
+    public void givenUserIsInitializedAndAdded(String name, int age, int id) {
+        this.userMock.setName(name);
+        this.userMock.setAge(age);
+        this.userMock.setId(id);
+        this.userService.addUser(userMock);
+    }
+    public void whenUserIsAddedSuccessfully() {
+        this.userService.addUser(userMock);
+    }
```

Comment:
at this point we dont know if user is being added successfully, as its a when method.